frontend/dialob-composer-material/src/dialogs/TextEditDialog.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import Markdown from 'react-markdown';
+import { Dialog, DialogTitle, DialogContent, Button, Typography, Box, TextareaAutosize } from '@mui/material';
+import { Visibility } from '@mui/icons-material';
+import { useEditor } from '../editor';
+import { FormattedMessage } from 'react-intl';
+import { useComposer } from '../dialob';
+import { DialogActionButtons, DialogHelpButton, DialogLanguageMenu } from './DialogComponents';
+
+const TextEditDialog: React.FC = () => {
+  const { updateItem } = useComposer();
+  const { editor, setTextEditDialogType, setActiveItem } = useEditor();
+  const item = editor.activeItem;
+  const open = editor.textEditDialogType !== undefined;
+  const [activeLanguage, setActiveLanguage] = React.useState(editor.activeFormLanguage);
+  const [preview, setPreview] = React.useState(false);
+  const [localizedText, setLocalizedText] = React.useState<string | undefined>(undefined);
+
+  const handleClose = () => {
+    setTextEditDialogType(undefined);
+    setActiveItem(undefined);
+  }
+
+  const handleClick = () => {
+    if (editor.textEditDialogType && item && localizedText) {
+      updateItem(item.id, editor.textEditDialogType, localizedText, activeLanguage);
+      handleClose();
+    }
+  }
+
+  React.useEffect(() => {
+    if (editor.textEditDialogType && item) {
+      setLocalizedText(item[editor.textEditDialogType]?.[activeLanguage]);
+    }
+  }, [activeLanguage, editor.textEditDialogType, item]);
+
+  if (!item) {
+    return null;
+  }
+
+  return (
+    <Dialog open={open} maxWidth='md' fullWidth>
+      <DialogTitle sx={{ display: 'flex', flexDirection: 'row', alignItems: 'center' }}>
+        <Typography>{item.id}: {editor.textEditDialogType}</Typography>
+        <Box flexGrow={1} />
+        <Box sx={{ display: 'flex', width: 0.35, justifyContent: 'space-between' }}>
+          <DialogLanguageMenu activeLanguage={activeLanguage} setActiveLanguage={setActiveLanguage} />
+          <Button color='inherit' endIcon={<Visibility />} onClick={() => setPreview(!preview)}><FormattedMessage id='preview' /></Button>
+          <DialogHelpButton helpUrl='https://www.markdownguide.org/basic-syntax/' />
+        </Box>
+      </DialogTitle>
+      <DialogContent>
+        {preview ?
+          <Box sx={{ border: 1, borderRadius: 0.5, borderColor: 'text.secondary' }}><Markdown>{localizedText}</Markdown></Box> :
```

Comment:
Try adding markdown component config for Material ui, like here:

https://github.com/dialob/dialob-parent/blob/dev/frontend/dialob-fill-material/src/components/MarkdownView.tsx

for better rendering within material.